src/watertap_contrib/reflo/costing/solar/flat_plate.py

```diff
@@ -0,0 +1,151 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2023, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+import pyomo.environ as pyo
+from watertap.costing.util import register_costing_parameter_block
+from watertap_contrib.reflo.costing.util import (
+    make_capital_cost_var,
+    make_fixed_operating_cost_var,
+    make_variable_operating_cost_var,
+)
+
+
+def build_flat_plate_cost_param_block(blk):
+
+    costing = blk.parent_block()
+
+    blk.cost_per_area_collector = pyo.Var(
+        initialize=600,
+        units=costing.base_currency / pyo.units.m**2,
+        bounds=(0, None),
+        doc="Cost per area for solar collector",
+    )
+
+    blk.cost_per_volume_storage = pyo.Var(
+        initialize=2000,
+        units=costing.base_currency / pyo.units.m**3,
+        bounds=(0, None),
+        doc="Cost per volume for thermal storage",
+    )
+
```

Comment:
Does the storage cost belong to the TES cost package? Or would it be double counted if the we also involve the TES component?

Reply:
This is a good point. Surrogate has built in storage but physical does not. I will have to implement a type check to make this work.